refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/HasTypeArgumentsTest.java

```diff
@@ -0,0 +1,83 @@
+package tech.picnic.errorprone.refaster.matchers;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.CompilationTestHelper;
+import com.google.errorprone.bugpatterns.BugChecker;
+import org.junit.jupiter.api.Test;
+
+final class HasTypeArgumentsTest {
+  @Test
+  void matches() {
+    CompilationTestHelper.newInstance(MatcherTestChecker.class, getClass())
+        .addSourceLines(
+            "A.java",
+            "import com.google.common.collect.ImmutableSet;",
+            "import java.util.ArrayList;",
+            "import java.util.List;",
+            "",
+            "class A {",
+            "  Object negative1() {",
+            "    return alwaysNull();",
+            "  }",
+            "",
+            "  Object negative2() {",
+            "    return new Object();",
+            "  }",
+            "",
+            "  List<Integer> negative3() {",
+            "    return new ArrayList<>();",
+            "  }",
+            "",
+            "  Foo negative4() {",
+            "    return new Foo(\"foo\");",
+            "  }",
+            "",
+            "  <E> ImmutableSet<E> positive1() {",
+            "    // BUG: Diagnostic contains:",
+            "    return ImmutableSet.<E>builder().build();",
```

Comment:
We can probably simplify by:

```
"  List<Integer> positive1() {",
"    return new ArrayList<Integer>();",
"  }",
"",
```

(Will apply this tomorrow when I get back to this.